test/integration/dss/test_dss_api.py

```diff
@@ -71,9 +83,31 @@ def test_python_nested_bundle_upload_download(self):
                 downloaded_file_paths = [object_name_builder(p, dest_dir) for p in downloaded_file_names]
                 self.assertEqual(uploaded_files.sort(), downloaded_file_paths.sort())
 
+    @unittest.skipIf(os.name is 'nt', 'Unable to test on Windows')  # TODO windows testing refactor
+    def test_python_exclamation_replacement(self):
+        bundle_path = os.path.join(TEST_DIR, "res", "nested_bundle")
+        uploaded_files = set(get_uploaded_file_names(bundle_path))
+
+        bundle_output = self.client.upload(src_dir=bundle_path,
+                                      replica="aws",
+                                      staging_bucket=self.staging_bucket)
+
+        bundle_uuid = bundle_output['bundle_uuid']
+        bundle_version = bundle_output['version']
+        bundle_fqid = "{}.{}".format(bundle_uuid, bundle_version)
+
+        manifest_files = bundle_output['files']
+        self.assertEqual({file['name'] for file in manifest_files}, uploaded_files)
+
+        with tempfile.TemporaryDirectory() as dest_dir:
+            self.client.download(bundle_uuid=bundle_output['bundle_uuid'], replica="aws", download_dir=dest_dir)
+            nested_downloaded_files = [file.name for file in os.scandir('{}/{}/zarr'.format(dest_dir, bundle_fqid))]
```

Comment:
Do you really have to list all files to assert that two of them were downloaded? What about `os.path.isfile`?